### DISCUSSION TITLE
程序异常退出，concurrent map read and map write

### DISCUSSION THREAD

**Discussion by hdfsun** [AlistGo/alist#2925] [Q&A]
运行环境：docker，linux_amd64，v3.7.2
程序异常退出
异常日志记录：

stderr: fatal error: concurrent map read and map write
stderr: 
stderr: goroutine 27643 [running]:
stderr: encoding/json.(*decodeState).object(0xc001721b90, {0x14f1ec0?, 0xc002a3fd70?, 0x0?})
stderr: 	/usr/lib/go/src/encoding/json/decode.go:703 +0x8c8
stderr: encoding/json.(*decodeState).value(0xc001721b90, {0x14f1ec0?, 0xc002a3fd70?, 0x90?})
stderr: 	/usr/lib/go/src/encoding/json/decode.go:374 +0x45
stderr: encoding/json.(*decodeState).unmarshal(0xc001721b90, {0x14f1ec0?, 0xc002a3fd70?})
stderr: 	/usr/lib/go/src/encoding/json/decode.go:181 +0x1de
stderr: encoding/json.Unmarshal({0xc0033fb500, 0x1048, 0x1500}, {0x14f1ec0, 0xc002a3fd70})
stderr: 	/usr/lib/go/src/encoding/json/decode.go:108 +0x125
stderr: github.com/go-resty/resty/v2.Unmarshalc(0xc0004c0000, {0xc00320e9b0, 0x10}, {0xc0033fb500, 0x1048, 0x1500}, {0x14f1ec0, 0xc002a3fd70})
stderr: 	/root/go/pkg/mod/github.com/go-resty/resty/v2@v2.7.0/util.go:109 +0x7a
stderr: github.com/go-resty/resty/v2.parseResponseBody(0xc0004c0000, 0xc000243f40)
stderr: 	/root/go/pkg/mod/github.com/go-resty/resty/v2@v2.7.0/middleware.go:333 +0x2ac
stderr: github.com/go-resty/resty/v2.(*Client).execute(0xc0004c0000, 0xc0035aa4e0)
stderr: 	/root/go/pkg/mod/github.com/go-resty/resty/v2@v2.7.0/client.go:972 +0xbda
stderr: github.com/go-resty/resty/v2.(*Request).Execute.func1()
stderr: 	/root/go/pkg/mod/github.com/go-resty/resty/v2@v2.7.0/request.go:769 +0x88
stderr: github.com/go-resty/resty/v2.Backoff(0xc0035972d0, {0xc0035972a8, 0x5, 0x0?})
stderr: 	/root/go/pkg/mod/github.com/go-resty/resty/v2@v2.7.0/retry.go:105 +0x19b
stderr: github.com/go-resty/resty/v2.(*Request).Execute(0xc0035aa4e0, {0x17a507d?, 0xc00320e950?}, {0x17fd4fa, 0x2f})
stderr: 	/root/go/pkg/mod/github.com/go-resty/resty/v2@v2.7.0/request.go:763 +0x537
stderr: github.com/go-resty/resty/v2.(*Request).Post(...)
stderr: 	/root/go/pkg/mod/github.com/go-resty/resty/v2@v2.7.0/request.go:699
stderr: github.com/alist-org/alist/v3/drivers/aliyundrive_share.(*AliyundriveShare).getFiles(0xc00096ef00, {0xc0033446f0, 0x28})
stderr: 	/app/drivers/aliyundrive_share/util.go:77 +0x8f4
stderr: github.com/alist-org/alist/v3/drivers/aliyundrive_share.(*AliyundriveShare).List(0xc0034cd75c?, {0x0?, 0xc00096ef00?}, {0x206f198?, 0xc002c3df80?}, {{0x1536c60?, 0xc000b18500?}})
stderr: 	/app/drivers/aliyundrive_share/driver.go:61 +0x3c
stderr: github.com/alist-org/alist/v3/internal/op.List.func1()
stderr: 	/app/internal/op/fs.go:98 +0x91
stderr: github.com/alist-org/alist/v3/pkg/singleflight.(*Group[...]).doCall.func2(0xc00359764e?, 0xc0006cc240, 0x7f32c5487f18)
stderr: 	/app/pkg/singleflight/singleflight.go:193 +0x71
stderr: github.com/alist-org/alist/v3/pkg/singleflight.(*Group[...]).doCall(0xc0006e9560?, 0xc0034cd9d0?, {0xc0034cd9d0?, 0x88b4ba?}, 0x206f228?)
stderr: 	/app/pkg/singleflight/singleflight.go:195 +0xb0
stderr: github.com/alist-org/alist/v3/pkg/singleflight.(*Group[...]).Do(0x2b3bfe0, {0xc0034cd9d0, 0x6d}, 0xc0035977c0)
stderr: 	/app/pkg/singleflight/singleflight.go:108 +0x176
stderr: github.com/alist-org/alist/v3/internal/op.List({0x206b1d0, 0xc002a3fcb0}, {0x2072e10?, 0xc00096ef00?}, {0xc0034cd75c, 0x41}, {{0xc0034cd730?, 0xc003597920?}}, {0x0, 0x0, ...})
stderr: 	/app/internal/op/fs.go:97 +0x433
stderr: github.com/alist-org/alist/v3/internal/fs.list({0x206b1d0, 0xc002a3fcb0}, {0xc0034cd730, 0x6d}, {0x0, 0x0, 0x0})
stderr: 	/app/internal/fs/list.go:25 +0x227
stderr: github.com/alist-org/alist/v3/internal/fs.List({0x206b1d0?, 0xc002a3fcb0?}, {0xc0034cd730, 0x6d}, {0x0?, 0xc000cf02a0?, 0x6d?})
stderr: 	/app/internal/fs/fs.go:17 +0x45
stderr: github.com/alist-org/alist/v3/internal/fs.WalkFS({0x206b1d0, 0xc00066ecf0}, 0xfffffffffffffffd, {0xc0034cd730, 0x6d}, {0x206f228, 0xc00290b800}, 0xc003597d70)
stderr: 	/app/internal/fs/walk.go:31 +0x198
stderr: github.com/alist-org/alist/v3/internal/fs.WalkFS({0x206b1d0, 0xc00066ecf0}, 0xfffffffffffffffe, {0xc001bff200, 0x5f}, {0x206f228, 0xc0024ca300}, 0xc003597d70)
stderr: 	/app/internal/fs/walk.go:37 +0x2b6
stderr: github.com/alist-org/alist/v3/internal/fs.WalkFS({0x206b1d0, 0xc00066ecf0}, 0xffffffffffffffff, {0xc0017221e0, 0x5d}, {0x206f228, 0xc001f284a0}, 0xc003597d70)
stderr: 	/app/internal/fs/walk.go:37 +0x2b6
stderr: github.com/alist-org/alist/v3/internal/search.BuildIndex({0x206b160?, 0xc0000c2000}, {0xc003597eb8, 0x1, 0xc001f284a0?}, {0xc000049400, 0x2, 0x2}, 0x6?, 0x0)
stderr: 	/app/internal/search/build.go:146 +0x4da
stderr: github.com/alist-org/alist/v3/internal/search.Update({0xc00247a210, 0x2c}, {0xc00029afd0, 0x1, 0xc0010817b8?})
stderr: 	/app/internal/search/build.go:219 +0x966
stderr: github.com/alist-org/alist/v3/internal/op.List.func1.1({0xc0001ed810?, 0x205b880?}, {0xc00029afd0, 0x1, 0x1})
stderr: 	/app/internal/op/fs.go:113 +0x6a
stderr: created by github.com/alist-org/alist/v3/internal/op.List.func1
stderr: 	/app/internal/op/fs.go:111 +0x2ad
stderr: 
stderr: goroutine 1 [chan receive, 72 minutes]:
stderr: github.com/alist-org/alist/v3/cmd.glob..func5(0x2a82360?, {0x17a55a9?, 0x1?, 0x1?})
stderr: 	/app/cmd/server.go:61 +0x385
stderr: github.com/spf13/cobra.(*Command).execute(0x2a82360, {0xc00029bc40, 0x1, 0x1})
stderr: 	/root/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:920 +0x847
stderr: github.com/spf13/cobra.(*Command).ExecuteC(0x2a82080)
stderr: 	/root/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:1044 +0x3bd
stderr: github.com/spf13/cobra.(*Command).Execute(...)
stderr: 	/root/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:968
stderr: github.com/alist-org/alist/v3/cmd.Execute()
stderr: 	/app/cmd/root.go:20 +0x25
stderr: main.main()
stderr: 	/app/main.go:6 +0x17
stderr: 
stderr: goroutine 19 [select, 3 minutes]:
stderr: github.com/Xhofe/go-cache.NewMemCache[...].func1()
stderr: 	/root/go/pkg/mod/github.com/!xhofe/go-cache@v0.0.0-20220723083548-714439c8af9a/cache.go:115 +0xf9
stderr: created by github.com/Xhofe/go-cache.NewMemCache[...]
stderr: 	/root/go/pkg/mod/github.com/!xhofe/go-cache@v0.0.0-20220723083548-714439c8af9a/cache.go:108 +0x28c
stderr: 
stderr: goroutine 20 [select, 3 minutes]:
stderr: github.com/Xhofe/go-cache.NewMemCache[...].func1()
stderr: 	/root/go/pkg/mod/github.com/!xhofe/go-cache@v0.0.0-20220723083548-714439c8af9a/cache.go:115 +0xf9
stderr: created by github.com/Xhofe/go-cache.NewMemCache[...]
stderr: 	/root/go/pkg/mod/github.com/!xhofe/go-cache@v0.0.0-20220723083548-714439c8af9a/cache.go:108 +0x28c
stderr: 
stderr: goroutine 21 [select, 3 minutes]:
stderr: github.com/Xhofe/go-cache.NewMemCache[...].func1()
stderr: 	/root/go/pkg/mod/github.com/!xhofe/go-cache@v0.0.0-20220723083548-714439c8af9a/cache.go:115 +0xf9
stderr: created by github.com/Xhofe/go-cache.NewMemCache[...]
stderr: 	/root/go/pkg/mod/github.com/!xhofe/go-cache@v0.0.0-20220723083548-714439c8af9a/cache.go:108 +0x28c
stderr: 
stderr: goroutine 22 [select, 3 minutes]:
stderr: github.com/Xhofe/go-cache.NewMemCache[...].func1()
stderr: 	/root/go/pkg/mod/github.com/!xhofe/go-cache@v0.0.0-20220723083548-714439c8af9a/cache.go:115 +0xf9
stderr: created by github.com/Xhofe/go-cache.NewMemCache[...]
stderr: 	/root/go/pkg/mod/github.com/!xhofe/go-cache@v0.0.0-20220723083548-714439c8af9a/cache.go:108 +0x28c
stderr:

**Comment by NovaShen0724** [comment]
并发读写错误，若重启无效考虑架构兼容性？
